### ISSUE TITLE
Allow use of None instead of 'None' string in Texture constructor

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#843]
### What should be added/changed?
**tl;dr** The constructor of `Texture` should be allowed to take an actual None value for the `hit_box_algorithm` argument.

At the moment, the `Texture` constructor takes a string `hit_box_algorithm` argument that only accepts string values. [Passing an actual None instead of the string `'None'` will cause a `TypeError` to be raised on line 127 of the texture constructor](https://github.com/pythonarcade/arcade/blob/daff5b3ca47138057d0002336a42448bf4f5fed5/arcade/texture.py#L127):
```py
        if hit_box_algorithm != "Simple" and \
           hit_box_algorithm != "Detailed" and \
           hit_box_algorithm != "None":
           raise ValueError("hit_box_algorithm must be 'Simple', 'Detailed', or 'None'.")
```

### What would it help with?

The framework will be more pythonic and intuitive for experienced python users while remaining accessible for absolute beginners who have never used python and don't understand what None is, and may try to put strings into the arguments.
